src/Cardano/Prelude/Base16.hs

```diff
@@ -30,3 +34,14 @@ parseBase16 s = do
   let (bs, suffix) = B16.decode $ Text.encodeUtf8 s
   unless (BS.null suffix) . Left $ Base16IncorrectSuffix suffix
   pure bs
+
+decodeEitherBase16 :: ByteString -> Either String ByteString
+decodeEitherBase16 bs = case B16.decode bs of
+  (decodedBs, suffix) -> if BS.null suffix
+    then Right decodedBs
+    else Left $ "invalid character at offset: " <> show (BS.length (BS.takeWhile isHex bs))
```

Comment:
Offset should be `BS.length decodedBs` I think.

Reply:
I implemented it that way to exactly emulate the behaviour of this function because eventually the intention is to change the implementation to delegate directly to it:

https://hackage.haskell.org/package/base16-bytestring-1.0.0.0/docs/Data-ByteString-Base16.html

Reply:
The examples from the docs were pasted into the `prop_decodeEitherBase16_examples` test and those pass at the moment.